File: backend/app/schemas/audit.py
from datetime import datetime

from pydantic import BaseModel


class AuditLogResponse(BaseModel):
    id: int
    organization_id: int | None
    actor_id: int
    action: str
    entity_type: str
    entity_id: int | None
    before: dict | None
    after: dict | None
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/audit.ts
export interface AuditLog {
  id: number
  organization_id: number | null
  actor_id: number
  actor_name: string
  action: string
  entity_type: string
  entity_id: number | null
  before: Record<string, unknown> | null
  after: Record<string, unknown> | null
  created_at: string
}
